predicators/envs/stick_button.py

```diff
@@ -611,14 +625,106 @@ def _get_tasks(self, num: int, num_button_lst: List[int],
+        rad = self.robot_radius
+        if not self.rz_x_lb + rad <= new_rx <= self.rz_x_ub - rad or \
+           not self.rz_y_lb + rad <= new_ry <= self.rz_y_ub - rad:
+            return state.copy()
+        next_state = state.copy()
+        next_state.set(self._robot, "x", new_rx)
+        next_state.set(self._robot, "y", new_ry)
+        next_state.set(self._robot, "theta", new_rtheta)
+        robot_circ = self.object_to_geom(self._robot, next_state)
+
+        # Check if the stick is held. If so, we need to move and rotate it.
+        stick_held = state.get(self._stick, "held") > 0.5
+        fingers_closed = state.get(self._robot, "fingers") <= 0.5
+        stick_rect = self.object_to_geom(self._stick, state)
+        assert isinstance(stick_rect, utils.Rectangle)
+        if stick_held and fingers_closed:
+            if not CFG.stick_button_disable_angles:
+                stick_rect = stick_rect.rotate_about_point(rx, ry, dtheta)
+            stick_rect = utils.Rectangle(x=(stick_rect.x + dx),
+                                         y=(stick_rect.y + dy),
+                                         width=stick_rect.width,
+                                         height=stick_rect.height,
+                                         theta=stick_rect.theta)
+            next_state.set(self._stick, "x", stick_rect.x)
+            next_state.set(self._stick, "y", stick_rect.y)
+            next_state.set(self._stick, "theta", stick_rect.theta)
+
+        if press > 0:
+            # Check if any button is now pressed.
+            tip_rect = self.stick_rect_to_tip_rect(stick_rect)
+            for button in state.get_objects(self._button_type):
+                circ = self.object_to_geom(button, state)
+                if (circ.intersects(tip_rect) and stick_held) or \
+                   (circ.intersects(robot_circ) and not stick_held):
+                    next_state.set(button, "pressed", 1.0)
+
+        if pickplace > 0:
+            # Check for placing the stick.
+            holder_rect = self.object_to_geom(self._holder, state)
+            if stick_held and fingers_closed and stick_rect.intersects(
+                    holder_rect):
+                # Place the stick back on the holder.
+                next_state.set(self._stick, "held", 0.0)
+                next_state.set(self._robot, "fingers", 1.0)
+
+            # Check if the stick is now held for the first time.
+            if not stick_held and stick_rect.intersects(robot_circ):
+                # Check for a collision with the stick holder. The reason that
+                # we only check for a collision here, as opposed to every
+                # timestep, is that we imagine the robot moving down in the z
+                # direction to pick up the stick, at which button it may
+                # collide with the stick holder. On other timesteps, the robot
+                # would be high enough above the holder to avoid collisions.
```

Comment:
the "at which button" part of this comment makes no sense to me... Could you update?

Reply:
Done.